portal/client-services/org.wso2.is.portal.user.client.api/src/main/java/org/wso2/is/portal/user/client/api/RecoveryMgtServiceImpl.java

```diff
@@ -273,15 +290,15 @@ public boolean verifyUsername(Map<String, String> userClaims) throws IdentityRec
      * Pass pass code value to persist
      *
      * @param uniqueUserId selected user id
-     * @param passCode generated one time password
      * @throws UserPortalUIException
      */
     @Override
-    public void persistPassCode(String uniqueUserId, String passCode) throws UserPortalUIException {
+    public void persistPassCode(String uniqueUserId) throws UserPortalUIException {
         try {
-            AdminForcePasswordResetManager.getInstance().persistPasscode(uniqueUserId, passCode);
+            AdminForcePasswordResetManager.getInstance().persistPasscode(uniqueUserId,
+                    passcodeGenerator.generatePasscode(MAX_LENGTH));
         } catch (IdentityRecoveryException e) {
-            throw new UserPortalUIException("Error while storing the Pass Code: " + passCode);
+            throw new UserPortalUIException("Error while storing the Pass Code");
```

Comment:
Since UserPortalUIException propagate to front end js, wrapping uniqueUserId is not necessary